src/apps/js_generic/js_generic.cpp

```diff
@@ -165,7 +165,12 @@ namespace ccfapp
     if (!buf)
       return JS_ThrowTypeError(ctx, "Argument must be an ArrayBuffer");
 
-    JSValue obj = JS_ParseJSON(ctx, (char*)buf, buf_size, "<json>");
+    std::vector<uint8_t> buf_null_terminated;
+    buf_null_terminated.reserve(buf_size + 1);
```

Comment:
```suggestion
    std::vector<uint8_t> buf_null_terminated(buf_size + 1);
```

Reply:
Also it's probably good to have a comment here to explain why we do this.

Reply:
I think this is slower as that constructor default-initializes all elements.